src/main/java/tw/commonground/backend/service/fact/FactService.java

```diff
@@ -0,0 +1,109 @@
+package tw.commonground.backend.service.fact;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import tw.commonground.backend.service.fact.dao.FactMapper;
+import tw.commonground.backend.service.fact.dao.FactRequest;
+import tw.commonground.backend.service.fact.dao.FactResponse;
+import tw.commonground.backend.service.fact.entity.FactEntity;
+import tw.commonground.backend.service.fact.entity.FactRepository;
+import tw.commonground.backend.service.reference.ReferenceEntity;
+import tw.commonground.backend.service.reference.ReferenceRepository;
+import tw.commonground.backend.service.reference.ReferenceRequest;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class FactService {
+
+    private final FactRepository factRepository;
+
+    private final ReferenceRepository referenceRepository;
+
+    private final FactMapper mapper = new FactMapper();
+
+    public FactService(FactRepository factRepository, ReferenceRepository referenceRepository) {
+        this.factRepository = factRepository;
+        this.referenceRepository = referenceRepository;
+    }
+
+    public List<FactResponse> getFacts(int page, int size, String column, String mode) {
+        Sort sorted = Sort.by((Objects.equals(mode, "asc") ? Sort.Order.by(column) : Sort.Order.desc(column)));
+        Pageable pageable = PageRequest.of(page, size, sorted);
+        Page<FactEntity> pageFacts = factRepository.findAll(pageable);
+        return pageFacts.getContent()
+                .stream()
+                .map(mapper::toResponse)
+                .collect(toList());
```

Comment:
Simply use `stream.toList()` here to create an immutable list, enhancing safety. Like:

```java
    .map(mapper::toResponse)
    .toList();
```

Reply:
Updated